src/field_theory/krull_topology.lean

```diff
@@ -283,3 +285,220 @@ begin
 end
 
 end totally_disconnected
+
+section compact
+
+variables {K L : Type*} [field K] [field L] [algebra K L] {E F : intermediate_field K L}
+
+/-- Let `E` be an intermediate field of `L/K` with `E/K` finite, and let `f` be an ultrafilter
+on `L ≃ₐ[K] L`. Then the restriction map `(L ≃ₐ[K] L) → (E →ₐ[K] L)` pushes `f` forward to an
+ultrafilter on `E →ₐ[K] L`. Since `E →ₐ[K] L` is a finite set, this ultrafilter is principal. The
+element of `E →ₐ[K] L` generating this principal ultrafilter is `f.alg_hom h_findim`, where
+`h_findim : finite_dimensional K E`. -/
+protected noncomputable def ultrafilter.generator_of_pushforward
+  {E : Type*} [field E] [algebra K E] [algebra E L] [is_scalar_tower K E L] [finite_dimensional K E]
```

Comment:
I made the change for the following lemma, but ran into trouble with the one after that, which requires the partial order on intermediate fields. 

I also did the definition of ultrafilter.glued_generators_of_pushforwards_function_spec using tactic mode. I'm sure it can easily be done term-style but I'm not sure how to - would you mind having a look?

Reply:
There's no reason to change the definition. You can look at the term which tactic mode has generated using `#print ultrafilter.glued_generators_of_pushforwards_function` and you can see that it's fine. A term mode definition would be something like

```lean
    let foo : finite_dimensional K (intermediate_field.adjoin K {x}) :=
      intermediate_field.adjoin.finite_dimensional (h_int x) in
    by exactI (f.generator_of_pushforward : (intermediate_field.adjoin K {x}) →ₐ[K] L)
  ⟨x, intermediate_field.mem_adjoin_simple_self K x⟩
```
but I think this is worse because then you get the `let` stuck in the term.

Reply:
I have gone for the term-style new definition, because Eric Wieser's previous suggested change is incompatible with the old definition. What do you think?